src/Events/Custom_Tables/V1/Migration/Events.php

```diff
@@ -51,6 +51,9 @@ public function get_ids_to_process( $limit ) {
 		// Batch locking
 		$batch_uid = uniqid( 'tec_ct1_action', true ); // Should be pretty unique.
 
+		// Let's avoid table locks in the following query, this could have Deadlock side effects.
+		$wpdb->query("SET SESSION TRANSACTION ISOLATION LEVEL READ UNCOMMITTED");
```

Comment:
This query, as any query, might fail.
I'm not sure what should be done if that is the case, and not collecting and looking at the return value here looks right to me, but I just want to make sure this is a conscious choice.

Also: add a `@see https://dev.mysql.com/doc/refman/5.7/en/set-transaction.html` to know what this is doing.

Reply:
Yeah, not sure we want to do anything else here. Don't think we'd want to fail out of this function - the only other option I can think of isn't a great one, but if this fails than to modify the following query to sidestep this potential lock. Also as far as I saw this syntax is valid for many versions of MySQL.

If you have other ideas let me know.

Reply:
No, fine as it is.